### DISCUSSION TITLE
Fine-tune ViT Models on Higher Resolution Images

### DISCUSSION THREAD

**Discussion by D0miH** [mlfoundations/open_clip#987] [Q&A]
Hi all, 
thank you so much for this awesome library!!

I am using the vision transformer CLIP models and I would like to test some properties with different number of patches.
The problem I have now is that there is only a limited number of models available with different patch sizes. For example, there are only the ViT-B/16 and the ViT-B/32 models, which have the same number of trainable parameters but different patch sizes.

Therefore, I would like to emulate different patch sizes by scaling the input images and fine-tuning the model. I know that the ViTs expect input images of size `224x224`. However, in the [original CLIP paper](https://arxiv.org/pdf/2010.11929) (sec. 3.2) they are using higher resolution images for fine-tuning. 

**So now to my question:**
1. Would it be possible to fine-tune the ViT-B/32 model with higher resolution images to emulate an arbitrary smaller batch size? 
2. How can I interpolate the positional embeddings? I have found a function for that [here](https://github.com/mlfoundations/open_clip/blob/49eac2f27a5bb98a7f7ecc1154918880aa55256c/src/open_clip/pos_embed.py#L75) within this repository. But as far as I can see, it is never used anywhere.

Thank you so much for your help!

**Comment by rwightman** [comment]
@D0miH if you pass `force_image_size` (cmd line `--force-image-size`) with the new image size to the model factories when you load pretrain weights it will interpolate for you... after that you should create a new config with that image size baked in. The train / eval scripts have that argument too and can be used to evaluate at a different resolution, or to fine-tune at a different resolution.

https://github.com/mlfoundations/open_clip/blob/49eac2f27a5bb98a7f7ecc1154918880aa55256c/src/open_clip/factory.py#L208-L253

**Comment by RuixiangZhao** [comment]
@D0miH Have you made any progress? I'm also looking to use high-resolution inputs for CLIP in the video-text retrieval task, but simply applying interpolation doesn't seem to improve performance. Specifically, I used the interpolation method provided [here](https://github.com/huggingface/transformers/blob/main/src/transformers/models/clip/modeling_clip.py#L200), where I input 320x320 frames into CLIP-B/32. However, after fine-tuning, the performance is still worse than the original 224x224 inputs. Do you have any suggestions?

**Comment by NimrodShabtay** [comment]
Hi @rwightman 
I am trying to use EVA-CLIP-B-16 pre-trained on 224x224 but inference on larger images.
I modified the config and used `force-image-size` when loading the model 
```
open_clip.create_model_and_transforms("EVA02-B-16", pretrained='merged2b_s8b_b131k',
                                                                force_image_size=(img_sz, img_sz))    
```

But I am getting an error on the shapes in the PatchEmbedding layer which is hard-coded initialized with 224
```
  File "xxxx/lib/python3.11/site-packages/timm/layers/patch_embed.py", line 116, in forward
    _assert(H == self.img_size[0], f"Input height ({H}) doesn't match model ({self.img_size[0]}).")
  File "xxxx/lib/python3.11/site-packages/torch/__init__.py", line 2132, in _assert
    assert condition, message
AssertionError: Input height (896) doesn't match model (224).
```
Any elegant way to workaround this issue?

Thanks
